src/permutation.jl

```diff
@@ -0,0 +1,65 @@
+export PermutationTest
+
+function ptstats(x,y,f)
+    xy = vcat(x,y)
+    rx = 1 : length(x)
+    ry = length(xy)-length(y)+1 : length(xy)
+    T = typeof(f(x))  # is there a better way?  e.g., does a funtype() exist, like eltype()??
```

Comment:
is there a function/macro that intputs another function and returns it's return type?  i would think that at least in some cases the type of returned variables could by inferred from the AST.  here i just called the function because i'm unware that this ability currently exists in julia.